### PR TITLE
feat: Adds support for rich content message editor in Campaigns

### LINKED ISSUE #2281: feat: Add support for rich content messages in Campaigns
Screenshot_2021-05-17 Chatwoot(1)

The current message field doesn't support rich content messages in campaigns. Use the rich content editor used in the message layout here as well.

### PR DESCRIPTION
**Pull request by iamsivin**
Fixes #2281 

https://user-images.githubusercontent.com/64252451/127969721-477c97f4-e73f-4345-b510-2daa281b6804.mov

### CODE REVIEW

**Comment by iamsivin** [comment]
Currently, the rich content editor doesn't support editing, we have to add that feature.

**Comment by iamsivin** [comment]
Due to some error, the text is not coming correctly inside the editor. I’ll create another pr to fix this

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
             v-model="message"
-            rows="5"
-            type="text"
+            class="editor"
+            :class="{ editor_warning: $v.message.$error }"
             :placeholder="$t('CAMPAIGN.ADD.FORM.MESSAGE.PLACEHOLDER')"
             @blur="$v.message.$touch"
           />
-          <span v-if="$v.message.$error" class="message">
+          <span v-if="$v.message.$error" class="warning-message">
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:28]: the `message` class is from foundation which already uses colors. Can you check if it works here ?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/CampaignsTable.vue [hunk 2]
```diff
@@ -168,6 +174,11 @@ export default {
       });
     },
   },
+  methods: {
+    convertedMessage(messages) {
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/CampaignsTable.vue:178]:
```suggestion
    convertedMessage(message) {
```

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:28]: I have checked it's not working here.

**Comment by pranavrajs** [comment]
@iamsivin @nithindavid If editing is not available now, then it would be a broken experience. Let us merge this only after the editing piece is fixed.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 3]
```diff
@@ -287,3 +293,29 @@ export default {
...
+.editor {
+  border: 1px solid var(--color-border);
+  border-radius: var(--border-radius-normal);
+  padding: 0 var(--space-slab);
+  margin-bottom: 0;
+}
+
+.editor_warning {
+  border: 1px solid var(--r-400);
+}
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:310]: Can we add this as a style to the editor component?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/CampaignsTable.vue [hunk 4]
```diff
@@ -224,6 +232,11 @@ export default {
       ];
     },
   },
+  methods: {
+    convertedMessage(message) {
+      return this.formatMessage(message);
+    },
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/CampaignsTable.vue:238]: Do we need this method? Can we directly call `formatMessage` in the JSX?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/EditCampaign.vue [hunk 5]
```diff
@@ -209,3 +213,25 @@ export default {
...
+
+.editor_warning {
+  border: 1px solid var(--r-400);
+}
+
+.warning-message {
+  color: var(--r-400);
+  font-weight: var(--font-weight-normal);
+  padding: var(--space-smaller) 0 0 0;
+}
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/EditCampaign.vue:236]: This seems to be repeated, please explore having this style in the editor component itself.

**Comment by muhsin-k** [comment]
The QA has been done in the local environment and everything is working as expected.


https://user-images.githubusercontent.com/12408980/127972473-5790d6df-3016-43d0-8947-1e1db398cfcf.mov

@pranavrajs / @sojan-official Do we need to enable rich editor for the `one_off` campaign?

**Comment by muhsin-k** [comment]
@iamsivin Let's disable the rich editor for `one_off` campaigns

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**